### PR TITLE
create signup page

### PR DESCRIPTION
**Pull request by tranhuyhoangbka**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+  
+  def create
+  	@user = User.new(user_params)
```
Comment (xuanduc987) [inline app/controllers/users_controller.rb]: remove () :)))

File: app/models/user.rb [hunk 2]
```diff
@@ -1,8 +1,13 @@
 class User < ActiveRecord::Base
+  before_save { self.email = email.downcase }
```
Comment (xuanduc987) [inline app/models/user.rb]: remove space after { and before }

Comment (tranhuyhoangbka) [inline app/models/user.rb]: yes

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (tranhuyhoangbka) [inline app/controllers/users_controller.rb]: no

File: app/models/user.rb (hunk 2 above)
Comment (tranhuyhoangbka) [inline app/models/user.rb]: thank

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (dieunb) [inline app/controllers/users_controller.rb]: indent

File: app/models/user.rb (hunk 2 above)
Comment (dieunb) [inline app/models/user.rb]: fix now

File: app/models/user.rb [hunk 3]
```diff
@@ -1,8 +1,13 @@
 class User < ActiveRecord::Base
+  before_save { self.email = email.downcase }	
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   has_many :learneds
   has_many :results
   has_many :activities
   validates :name, presence: true, length: {maximum: 60}
-  validates :email, presence: true, length: {maximum: 80}
-  validates :password_digest, presence: true, length: {maximum: 80}
+  validates :email, presence: true, length: {maximum: 80}, format: { with: VALID_EMAIL_REGEX },
```
Comment (dieunb) [inline app/models/user.rb]:
```
format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  def new
+  	@user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  def new
+  	@user = User.new
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: indent 2

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
+  	if @user.save      
+      flash[:success] = "wellcome E-Learning System"
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      redirect_to @user
+  	else
+  	  render 'new'
+  	end
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    if @user.save      
+      flash[:success] = "wellcome E-Learning System"
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save      
+      flash[:success] = "wellcome E-Learning System"
+      redirect_to @user
+    else
+      render 'new'
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: "

File: app/views/users/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,22 @@
+<%= provide :title, 'Sign up' %>
+<h1>Sign up</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+  <%= form_for @user do |f| %>
+    <%= render 'shared/error_messages', object: f.object %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: "

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save      
+      flash[:success] = 'wellcome E-Learning System'
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: "

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+      redirect_to @user
+    else
+      render 'new'
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: "

**Pull request merged**